### PR TITLE
ARROW-12617: [Python] Align orc.write_table keyword order with parquet.write_table

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @mathyingzhou

File: python/pyarrow/orc.py [hunk 1]
```diff
@@ -197,6 +198,13 @@ def write_table(where, table):
     table : pyarrow.lib.Table
         The table to be written into the ORC file
```
Comment (pitrou) [inline python/pyarrow/orc.py:199]: Can you reorder the docstring as well? :-)

File: python/pyarrow/orc.py [hunk 2]
```diff
@@ -197,6 +198,13 @@ def write_table(where, table):
...
     """
+    if isinstance(where, Table):
+        warnings.warn(
+            "The order of the keyword arguments has changed. Pass as "
```
Comment (pitrou) [inline python/pyarrow/orc.py:203]: "The order of arguments". Keyword arguments are unrelated.

File: python/pyarrow/orc.py (hunk 1 above)
Comment (jorisvandenbossche) [reply python/pyarrow/orc.py:199]: Ah, yes ;)

**Review by pitrou — APPROVED** [review]

**Pull request closed without merging**